### PR TITLE
Fix reconciler controllers to exit cleanly

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

File: pkg/remediator/reconcile/worker_test.go [hunk 1]
```diff
@@ -126,6 +127,55 @@ func TestWorker_ProcessNextObject(t *testing.T) {
 	}
 }
 
+func TestWorker_Run_Cancelled(t *testing.T) {
+	q := queue.New("test")
+
+	toProcess := []client.Object{}
```
Comment (sdowell) [inline pkg/remediator/reconcile/worker_test.go:133]: Seems like all of for loops are a noop since this is empty

**Review by sdowell — COMMENTED** [review]
Functional changes LGTM but it seems like the unit testing is incomplete

File: pkg/remediator/reconcile/worker_test.go (hunk 1 above)
Comment (karlkfi) [reply pkg/remediator/reconcile/worker_test.go:133]: Yeah, I copied most of this from another test. I wasn't sure if I should add more tests or simplify this one.

Comment (karlkfi) [reply pkg/remediator/reconcile/worker_test.go]: Alright, I cleaned this one up and added another to test a queue with items still left in it.

**Pull request merged**